framework/src/postprocessors/ComparisonPostprocessor.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComparisonPostprocessor.h"
+
+template <>
+InputParameters
+validParams<ComparisonPostprocessor>()
+{
+  InputParameters params = validParams<GeneralPostprocessor>();
+
+  MooseEnum comparison_type("equals greater_than_equals less_than_equals greater_than less_than");
+  params.addRequiredParam<MooseEnum>("comparison_type",
+                                     comparison_type,
+                                     "The type of comparison to perform. Options are: " +
+                                         comparison_type.getRawNames());
+
+  params.addParam<Real>("absolute_tolerance",
+                        libMesh::TOLERANCE * libMesh::TOLERANCE,
+                        "Absolute tolerance used in comparisons");
+
+  return params;
+}
+
+ComparisonPostprocessor::ComparisonPostprocessor(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+
+    _comparison_type(getParam<MooseEnum>("comparison_type")),
+    _absolute_tolerance(getParam<Real>("absolute_tolerance"))
+{
+}
+
+bool
+ComparisonPostprocessor::comparisonIsTrue(const Real & a, const Real & b) const
+{
+  if (_comparison_type == "equals")
```

Comment:
Since this gets called on every element or node, I'd switch to a real enum fed from a MooseEnum. See the NumDofs postprocessor for how that works. String comparisons can be quite expensive.

Reply:
This PPS inherits from `GeneralPostprocessor`...